src/PhpSpec/Process/Shutdown/Shutdown.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Process\Shutdown;
+
+final class Shutdown
+{
+    protected $actions;
+
+    const HHVM_FATAL_ERROR = 16777217;
+
+    public function __construct()
+    {
+        $this->actions = array();
+    }
+
+    public function registerShutdown()
+    {
+        error_reporting(E_ERROR);
+        register_shutdown_function(array($this, 'runShutdown'));
+    }
+
+    public function registerAction(ShutdownActionInterface $action)
+    {
+        $this->actions[] = $action;
+    }
+
+    public function runShutdown()
+    {
+        $error = $this->getFatalError();
+
+        foreach ($this->actions as $fatalErrorActions) {
+            $fatalErrorActions->runAction($error);
+        }
+    }
+
+    private function getFatalError()
+    {
+        $error = error_get_last();
+        $fatal = false;
+
+        if (!empty($error)) {
+            $fatal = isset($_ENV['HHVM']) ? (self::HHVM_FATAL_ERROR == $error['type']) : (E_ERROR == $error['type']);
```

Comment:
why checking in ``$_ENV['HHVM']`` ? ``defined('HHVM_VERSION')`` is the way to detect that you are running on HHVM, not an environment variable being set